nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "        per_device_train_batch_size=4,\n",
+    "        gradient_accumulation_steps=2,\n",
+    "        per_device_eval_batch_size=4,\n",
+    "        evaluation_strategy=\"steps\",\n",
+    "        eval_steps=100, \n",
+    "        disable_tqdm=False,\n",
+    "        load_best_model_at_end=True,\n",
+    "        metric_for_best_model='accuracy', # default value is validation loss, we want the model with the highest accuracy \n",
+    "        warmup_steps=200,\n",
+    "        logging_steps=100,\n",
+    "        learning_rate=5e-6,\n",
+    "        fp16=True,\n",
+    "        dataloader_num_workers=0,\n",
+    "        output_dir='./{}-classification-consumer-complaints-results'.format(\n",
+    "            identifier),\n",
+    "        run_name='{}-classification-consumer-complaints'.format(identifier),\n",
+    "        logging_dir='./{}-classification-consumer-complaints-logging'.format(\n",
+    "            identifier),\n",
+    "        save_strategy=\"steps\"\n",
+    "    )\n",
+    "\n",
+    "    trainer = Trainer(\n",
+    "        model=model,\n",
+    "        args=training_args,\n",
+    "        compute_metrics=compute_metrics,\n",
+    "        train_dataset=train_ds,\n",
+    "        eval_dataset=test_ds,\n",
+    "    )\n",
+    "    start_time = time.time()\n",
+    "    trainer.train()\n",
+    "    duration = time.time() - start_time\n",
+    "\n",
+    "    metrics = trainer.evaluate()\n",
+    "\n",
+    "    trainer.save_model(f\"models/{identifier}\")\n",
+    "    tokenizer.save_pretrained(f\"models/{identifier}\")\n",
+    "\n",
+    "    return trainer, parameters, metrics, duration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "id": "_VomBSwDejiy"
+   },
+   "outputs": [],
+   "source": [
+    "token_count_cutoff = IMDBDatasetTokenCountType.LONG # Run for both SHORT and LONG scenario"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "1JYNVDC8BSZW",
+    "outputId": "0ad30f96-d5b9-4ec9-d3a9-8aa3ee3683b9"
+   },
+   "outputs": [],
+   "source": [
+    "train_ds, test_ds = filter_train_test_dataset_with_tokencount(ds['train'], ds['test'], token_count_cutoff)\n",
+    "# To fairly compare both scenarios, we want to limit the size of both the training and test set to the most most restricting size. \n",
+    "max_train_samples = 1270\n",
+    "max_test_samples = 1186\n",
+    "\n",
+    "# shuffle datasets to avoid imbalanced data issues during training/evaluation\n",
+    "train_ds = train_ds.shuffle(seed=42).select(range(max_train_samples))\n",
+    "test_ds = test_ds.shuffle(seed=42).select(range(max_test_samples))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "id": "t43_73glBSZU"
+   },
+   "outputs": [],
+   "source": [
+    "def get_deep_size(obj, seen=None):\n",
```

Comment:
describe this method a bit (eg in a text cell above): what is this, why do we want to do this

Reply:
Totally agree! Will include a description 👌